modules/plugin-search.php

```diff
@@ -0,0 +1,486 @@
+
+		if ( false === $data || is_wp_error( $data ) ) {
+			include_once( ABSPATH . 'wp-admin/includes/plugin-install.php' );
+			$data = plugins_api( 'plugin_information', array(
+				'slug' => 'jetpack',
+				'is_ssl' => is_ssl(),
+				'fields' => array(
+					'banners' => true,
+					'reviews' => true,
+					'active_installs' => true,
+					'versions' => false,
+					'sections' => false,
+				),
+			) );
+			set_transient( 'jetpack_plugin_data', $data, DAY_IN_SECONDS );
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Create a list with additional features for those we don't have a module, like Akismet.
+	 *
+	 * @since 7.1.0
+	 *
+	 * @return array List of features.
+	 */
+	public function get_extra_features() {
+		return array(
+			'akismet' => array(
+				'name' => 'Akismet',
+				'search_terms' => 'akismet, spam, block',
+				'short_description' => esc_html__( 'Prevent spam and protect you and your site from malicious content.', 'jetpack' ),
+				'requires_connection' => true,
+				'module' => 'akismet',
+				'sort' => '16',
+				'learn_more_button' => 'https://jetpack.com/features/security/spam-filtering/'
+			),
+		);
+	}
+
+	/**
+	 * Intercept the plugins API response and add in an appropriate card for Jetpack
+	 */
+	public function inject_jetpack_module_suggestion( $result, $action, $args ) {
+		require_once JETPACK__PLUGIN_DIR . 'class.jetpack-admin.php';
+		$jetpack_modules_list = Jetpack_Admin::init()->get_modules();
+
+		// Never suggest this module.
+		unset( $jetpack_modules_list['plugin-search'] );
+
+		// Looks like a search query; it's matching time
+		if ( ! empty( $args->search ) ) {
+			$matching_module = null;
+
+			// Record event when user searches for a term
+			JetpackTracking::record_user_event( 'wpa_plugin_search_term', array( 'search_term' => $args->search ) );
+
+			// Lowercase, trim, remove punctuation/special chars, decode url, remove 'jetpack'
+			$normalized_term = $this->sanitize_search_term( $args->search );
+
+			$jetpack_modules_list = array_merge( $this->get_extra_features(), $jetpack_modules_list );
+
+			usort( $jetpack_modules_list, array( $this, 'by_sorting_option' ) );
+
+			// Try to match a passed search term with module's search terms
+			foreach ( $jetpack_modules_list as $module_slug => $module_opts ) {
+				$terms_array = explode( ', ', strtolower( $module_opts['search_terms'] . ', ' . $module_opts['name'] ) );
+				if ( in_array( $normalized_term, $terms_array ) ) {
+					$matching_module = $module_slug;
+					break;
+				}
+			}
+
+			if ( isset( $matching_module ) && $this->is_not_dismissed( $jetpack_modules_list[ $matching_module ]['module'] ) ) {
+				// Record event when a matching feature is found
+				JetpackTracking::record_user_event( 'wpa_plugin_search_match_found', array( 'feature' => $matching_module ) );
+
+				$inject = (array) self::get_jetpack_plugin_data();
+				$image_url = plugins_url( 'modules/plugin-search/psh', JETPACK__PLUGIN_FILE );
```

Comment:
Given that it's a fairly generic image, I wonder if we should add it to the images directory instead; maybe we can re-use it for other things in the future?